### PR TITLE
Move the flake at the root of the repository

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This change simplifies the nix usage so that we can run:

  nix run github:haskell/security-advisories

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -16,7 +16,7 @@
           pkgs.haskellPackages.developPackage {
             inherit returnShellEnv;
             name = "hsec-tools";
-            root = ./.;
+            root = "${self}/code/hsec-tools";
```
Comment (blackheaven) [inline flake.nix:19]: `./code/hsec-tools` does not work?

**Review by blackheaven — APPROVED** [review]

**Review by blackheaven — CHANGES_REQUESTED** [review]
There's also `.envrc` and `shell.nix`

**Comment by akacase** [comment]
@TristanCacqueray while you're changing those we can reduce our usage on the `shell.nix` compat piece by just using builtins now via: `(builtins.getFlake ("git+file://" + toString ./.)).devShells.${builtins.currentSystem}.default`

File: flake.nix (hunk 1 above)
Comment (TristanCacqueray) [reply flake.nix:19]: That works, TIL :-)

**Comment by TristanCacqueray** [comment]
Thanks for the quick review! @akacase your shell.nix didn't work (error: attribute 'devShells' missing), so I modified it a bit.

**Review by blackheaven — APPROVED** [review]

**Review by akacase — APPROVED** [review]

**Pull request merged**